### PR TITLE
Only check latest release version number on GitHub every 10 minutes

### PR DESCRIPTION
**Pull request by ethauvin**
See https://github.com/cbeust/kobalt/issues/373. Please review.

One could argue that the timing should be configurable, or at least overwritable. Thoughts?

### CODE REVIEW

File: modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt [hunk 1]
```diff
@@ -110,38 +114,42 @@ class GithubApi2 @Inject constructor(
         get() {
             val callable = Callable<String> {
                 var result = "0"
```
Comment (cbeust) [inline modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt]: While you're at it, I strongly suspect this is why we sometimes find a `0` in `kobalt-wrapper.properties` when something goes wrong. A more sensible default would be the same version that was in the original file.

But if you're going to fix this, do it in a separate PR.

File: modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt [hunk 2]
```diff
@@ -110,38 +114,42 @@ class GithubApi2 @Inject constructor(
...
-                        warn("Couldn't call Github.getReleases(): $jsonError")
-                    } else {
-                        val releases = ex.body()
-                        if (releases != null) {
-                            releases.firstOrNull()?.let {
-                                try {
-                                    result = listOf(it.name, it.tagName).filterNotNull().first { !it.isBlank() }
-                                } catch(ex: NoSuchElementException) {
-                                    throw KobaltException("Couldn't find the latest release")
+                if (Duration.ofMinutes(10L) >
```
Comment (cbeust) [inline modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt]: Might be a bit annoying for me and other Kobalt developers, but we can look into making this configurable later.

File: modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt (hunk 1 above)
Comment (ethauvin) [inline modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt]: I should've have seen that. I'll PR the fix.

File: modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt (hunk 2 above)
Comment (ethauvin) [inline modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt]: How about using the `-dev` parameter to disable it?

Comment (cbeust) [inline modules/kobalt-plugin-api/src/main/kotlin/com/beust/kobalt/misc/GithubApi2.kt]: Brilliant idea.
